allennlp/modules/token_embedders/embedding.py

```diff
@@ -140,6 +149,25 @@ def forward(self, inputs):  # pylint: disable=arguments-differ
             embedded = projection(embedded)
         return embedded
 
+    @overrides
+    def extend_vocab(self, extended_vocab: Vocabulary, vocab_namespace: Optional[str] = None):
+        if not vocab_namespace:
+            vocab_namespace = getattr(self, "_vocab_namespace", None)
+
+        if not vocab_namespace:
+            vocab_namespace = "tokens"
+            logging.warning("Neither original training store vocab_namespace for Embedding class "
+                            "nor user passed vocab_namespace in extended_vocab. So, "
+                            "defaulting to 'tokens' namespace for embedding extension.")
+
+        extended_num_embeddings = extended_vocab.get_vocab_size(vocab_namespace)
+        extra_num_embeddings = extended_num_embeddings - self.num_embeddings
+        embedding_dim = self.weight.data.shape[-1]
+        extra_weight = torch.FloatTensor(extra_num_embeddings, embedding_dim)
+        torch.nn.init.xavier_uniform_(extra_weight)
```

Comment:
You should document somewhere (in this method's docstring?) that you're initializing the new rows with `xavier_uniform`.

Reply:
Sure.